plugins/chat/lib/chat_message_bookmarkable.rb

```diff
@@ -31,7 +31,7 @@ def self.search_query(bookmarks, query, ts_query, &bookmarkable_search)
   end
 
   def self.validate_before_create(guardian, bookmarkable)
-    if bookmarkable.blank? || !guardian.can_see_chat_channel?(bookmarkable.chat_channel)
+    if bookmarkable.blank? || !guardian.can_join_chat_channel?(bookmarkable.chat_channel)
```

Comment:
no tests need to change or be created for this?

Reply:
Because guardian is tested, I added a test which ensures that guardian is invoked